doc/source/howtos/encrypt-traffic.rst

```diff
@@ -0,0 +1,319 @@
+  http://dev.mysql.com/doc/refman/5.6/en/creating-ssl-files-using-openssl.html
+
+  The following procedure shows how to generate the :file:`ca.pem`,
+  :file:`server-cert.pem`, and :file:`server-key.pem` files.
+
+  1. To generate the CA certificate::
+
+      openssl genrsa 2048 > ca-key.pem
+      openssl req -new -x509 -nodes -days 3600 -key ca-key.pem -out ca.pem
+
+  #. To generate the server certificate, remove passphrase, and sign it::
+
+      openssl req -newkey rsa:2048 -days 3600 -nodes -keyout server-key.pem -out server-req.pem
+      openssl rsa -in server-key.pem -out server-key.pem
+      openssl x509 -req -in server-req.pem -days 3600 -CA ca.pem -CAkey ca-key.pem -set_serial -1 -out server-cert.pem
+
+  #. (Optional) To generate the client certificate,
+     remove passphrase, and sign it::
+
+      openssl req -newkey rsa:2048 -days 3600 -nodes -keyout client-key.pem -out client-req.pem
+      openssl rsa -in client-key.pem -out client-key.pem
+      openssl x509 -req -in client-req.pem -days 3600 -CA ca.pem -CAkey ca-key.pem -set_serial 01 -out client-cert.pem
+
+  There are two ways to deploy SSL files across the cluster:
+
+  * (Preferred) Use the same files for all machines in the configuration.
+    Generate the files on one machine (or use the MySQL-generated files),
+    and copy the files to each node in the cluster.
+    Follow steps (1) and (2) in the previous procedure
+    to manually generate the files.
+
+  * (Alternative) Generate one CA file with separate server certificates
+    signed by that one CA file.
+    Do step (1) to generate the CA file
+    and then do step (2) for each server using the same CA file.
+    So each server will have different :file:`server-cert.pem` files,
+    but they will all share the same :file:`ca.pem` file.
+
+  .. note:: Whatever method you use to generate the certificate and key files,
+     the ``Common Name`` value used for the server and client certificates/keys
+     must each differ from that value used for the CA certificate.
+     Otherwise, the certificate and key files will not work for servers
+     compiled using OpenSSL.
+
+     The easiest way to do this is to give the CA file a common name (CN)
+     as follows::
+
+      openssl req -new -x509 -nodes -days 3600 -key ca-key.pem -out ca.pem -subj “/CN=my_cluster_name”
+
+.. note:: SSL clients require DH parameters to be at least 1024 bits,
+   due to the `logjam vulnearability
+   <https://en.wikipedia.org/wiki/Logjam_(computer_security)>`.
+   However, versions of ``socat`` earlier than 1.7.3 use 512-bit parameters.
+   If a :file:`dhparams.pem` file of required length is not found during SST
+   in the data directory,
+   it is generated with 2048 bits, which can take several minutes.
+   To avoid this delay, create the :file:`dhparams.pem` file manually
+   and place it in the data directory before joining the node to the cluster.
+
+IST Traffic, Write-Set Replication, and Service Messages
```

Comment:
For consistency, shouldn't this be "Securing IST Traffic, Write-Set Replication, and Service Messages"